python/pyarrow/tests/test_serialization.py

```diff
@@ -140,6 +152,13 @@ def assert_equal(obj1, obj2):
 ]
 
 
+# index_types = ('i1', 'i2', 'i4', 'i8', 'u1', 'u2', 'u4', 'u8')
+# tensor_types = ('i1', 'i2', 'i4', 'i8', 'u1', 'u2', 'u4', 'u8',
+#                 'f2', 'f4', 'f8')
+index_types = ('i8',)
+tensor_types = ('i8',)
```

Comment:
@pitrou @mrkn this mostly works now, but if I enable all datatypes here for serialization roundtrip tests, then roundtrip tests segfault **only** on xcode clang build (e.g.: https://travis-ci.org/apache/arrow/jobs/603260438#L4045). Other builds seem to be ok. I'm guessing it's maybe a borrowed reference issue, but I'm a bit stumped.

Reply:
This seems to be have been fixed now.